Src/AutoFixture/DataAnnotations/EnumDataTypeAttributeRelay.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using AutoFixture.Kernel;
+
+namespace AutoFixture.DataAnnotations
+{
+    /// <summary>
+    /// Handles a request for a string that matches an Enum data type.
+    /// </summary>
+    public class EnumDataTypeAttributeRelay : ISpecimenBuilder
+    {
+        private readonly object syncLock = new object();
+        private readonly Dictionary<Type, IEnumerator> enumerators = new Dictionary<Type, IEnumerator>();
+        private IRequestMemberTypeResolver requestMemberTypeResolver = new RequestMemberTypeResolver();
+
+        /// <summary>
+        /// Gets or sets the current IRequestMemberTypeResolver.
+        /// </summary>
+        public IRequestMemberTypeResolver RequestMemberTypeResolver
+        {
+            get => this.requestMemberTypeResolver;
+            set => this.requestMemberTypeResolver = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        /// <summary>
+        /// Creates a new specimen based on a request.
+        /// </summary>
+        /// <param name="request">The request that describes what to create.</param>
+        /// <param name="context">A context that can be used to create other specimens.</param>
+        /// <returns>
+        /// The requested specimen if possible; otherwise a <see cref="NoSpecimen"/> instance.
+        /// </returns>
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (request == null)
+            {
+                return new NoSpecimen();
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var enumDataTypeAttribute = TypeEnvy.GetAttribute<EnumDataTypeAttribute>(request);
+            if (enumDataTypeAttribute == null)
+            {
+                return new NoSpecimen();
+            }
+
+            if (!this.RequestMemberTypeResolver.TryGetMemberType(request, out var memberType))
+            {
+                return new NoSpecimen();
+            }
+
+            if (!enumDataTypeAttribute.EnumType.GetTypeInfo().IsEnum)
+                return new NoSpecimen();
+
+            if (memberType == typeof(string))
```

Comment:
I have added validating of byte, sbyte, short, ushort, int, uint, long and ulong types.

> We should also account for the underlying type of the `enum`.

Could you explain how to implement this correctly?

Reply:
@elasskiy tbh I can't really remember what I meant by that comment. 😅  
However I've also added support for `object` and the actual `enum` type. According to [this](https://github.com/microsoft/referencesource/blob/master/System.ComponentModel.DataAnnotations/DataAnnotations/EnumDataTypeAttribute.cs) implementation these are also valid scenarios, and since we've already generated the value...